RabinKarpAlgorithm/RabinKarp.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RabinKarpAlgorithm
+{
+    public class RabinKarp
+    {
+        int mod = 113;
+        public int GetPatternCount(string baseString, string pattern)
+        {
+            int baseStringLen = baseString.Length;
+            int patternLen = pattern.Length;
+
+            char[] possibleMatchChar = new char[patternLen];
+
+            int patternHash = HashMethod(pattern);
+
+            int cnt = 0;
+            int sum = 0;
+
+            int i, j = 0;
+            int pwr = patternLen - 1;
+
+            bool match = false;
+
+            while (j < patternLen)
+            {
+                possibleMatchChar[j] = baseString[j];
+                j++;
+            }
+
+            sum = HashMethod(new string(possibleMatchChar));
+            if (sum == patternHash)
+                match = pattern.Equals(new string(possibleMatchChar));
+            if (match == true)
+                cnt++;
+
+            for (i = 1; i <= baseStringLen - patternLen; i++)
+            {
+                j = 0;
+                match = false;
+
+                while (j < patternLen - 1)
```

Comment:
Cemu ovaj while? ako hoces da trimujes string postoje metode i za to. Nemoj to da radis rucno, nego pozovi metodu koja ce to da ti odradi.

Reply:
Jasno, u sustini samo ubacuje prvih patternLen karaktera u niz karaktera za koji treba racunati hash vrednost. Kontam da Trim moze to da odradi u jednoj liniji.